### PR TITLE
Israel.c/jooq api + service & tests 

### PR DESCRIPTION
**Pull request by IsraelCohen54**
api + service & tests
** Only last 2 functions inside controller are relevant ***

### CODE REVIEW

File: src/main/kotlin/com/hibob/academy/resource/ex10-controler.kt [hunk 1]
```diff
@@ -1,12 +1,19 @@
...
+import org.springframework.beans.factory.annotation.Autowired
+
 
-@Path("/pets")
-class PetController {
+@Path("/api")
+class DaoController {
+
+    @Autowired
+    private lateinit var petDao: PetDao
```
Comment (IsraelCohen54) [inline src/main/kotlin/com/hibob/academy/resource/ex10-controler.kt]: Please remove

File: src/main/kotlin/com/hibob/academy/resource/ex10-controler.kt [hunk 2]
```diff
@@ -1,12 +1,19 @@
...
+    private val petService:PetService = PetService(petDao)
```
Comment (IsraelCohen54) [inline src/main/kotlin/com/hibob/academy/resource/ex10-controler.kt]: No need to init

File: src/main/kotlin/com/hibob/academy/service/PetService.kt [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import com.hibob.academy.dao.PetDao
+import org.springframework.stereotype.Service
+
+@Service
+class PetService(private val petDao: PetDao) {
+
+    fun adoptPet(petId: Long, ownerId: Long, companyId: Long) {
+        val currentOwnerId = petDao.getPetOwnerId(petId)
+        currentOwnerId?.let {
+            throw IllegalStateException()
```
Comment (IsraelCohen54) [inline src/main/kotlin/com/hibob/academy/service/PetService.kt]: Add message to exception

File: src/test/kotlin/com/hibob/academy/service/PetServiceTest.kt [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+import org.junit.jupiter.api.Assertions.assertThrows
+import org.junit.jupiter.api.Test
+import org.mockito.Mockito.*
+import org.mockito.kotlin.any
+import org.mockito.kotlin.eq
+import org.mockito.kotlin.whenever
+
+class PetServiceTest {
+
+    private val petDao: PetDao = mock(PetDao::class.java)
```
Comment (IsraelCohen54) [inline src/test/kotlin/com/hibob/academy/service/PetServiceTest.kt]: mock {}

File: src/main/kotlin/com/hibob/academy/service/PetService.kt (hunk 3 above)
Comment (IsraelCohen54) [reply src/main/kotlin/com/hibob/academy/service/PetService.kt]: done

**Review by omermelamed — APPROVED** [review]

**Pull request merged**